src/main/java/org/betonquest/betonquest/database/AsyncSaver.java

```diff
@@ -62,12 +63,19 @@ public void run() {
                         active = false;
                         wait();
                     } catch (final InterruptedException e) {
-                        log.error("There was a exception with SQL", e);
+                        log.warn("AsyncSaver got interrupted!");
                     }
                 }
             }
             if (!active) {
-                con.refresh();
+                while (!con.refresh()) {
+                    log.warn("Failed to re-establish connection with the database! Trying again in one second...");
+                    try {
+                        sleep(1000L);
```

Comment:
This values should be configurable or at least a constant instead of a magic value. How about adding it to the `mysql` config section as `reconnect_interval`?